clusters/resource_cluster.go

```diff
@@ -57,71 +58,116 @@ func ZoneDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 	return false
 }
 
-func resourceClusterSchema() map[string]*schema.Schema {
-	return common.StructToSchema(Cluster{}, func(s map[string]*schema.Schema) map[string]*schema.Schema {
-		s["spark_conf"].DiffSuppressFunc = SparkConfDiffSuppressFunc
-		common.MustSchemaPath(s, "aws_attributes", "zone_id").DiffSuppressFunc = ZoneDiffSuppress
-		common.MustSchemaPath(s, "gcp_attributes", "use_preemptible_executors").Deprecated = "Please use 'availability' instead."
+type ClusterResourceProvider struct{}
 
-		common.MustSchemaPath(s, "init_scripts", "dbfs").Deprecated = DbfsDeprecationWarning
+func (ClusterResourceProvider) UnderlyingType() compute.ClusterDetails {
+	return compute.ClusterDetails{}
+}
 
-		// adds `library` configuration block
-		s["library"] = common.StructToSchema(libraries.ClusterLibraryList{},
-			func(ss map[string]*schema.Schema) map[string]*schema.Schema {
-				ss["library"].Set = func(i any) int {
-					lib := libraries.NewLibraryFromInstanceState(i)
-					return schema.HashString(lib.String())
-				}
-				return ss
-			})["library"]
+func (ClusterResourceProvider) Aliases() map[string]string {
+	return map[string]string{"cluster_mount_infos": "cluster_mount_info"}
+}
 
-		s["autotermination_minutes"].Default = 60
-		s["cluster_id"] = &schema.Schema{
+func (ClusterResourceProvider) CustomizeSchema(s map[string]*schema.Schema) map[string]*schema.Schema {
+	common.CustomizeSchemaPath(s, "enable_elastic_disk").SetOptional().SetComputed()
```

Comment:
This looks a bit difficult to maintain and I think something similar would be required for jobs resource as well? should we split this up so it's easier to read, for example, we could split on top level field, for example `init_scripts`, `workload_type`? A question - how do we check if customizations are complete?

Reply:
Yeah for the jobs resource, we would just set whatever path that leads to the cluster reference to be the schema derived from `ClusterResourceProvider` so we don't have to duplicate it.

For making sure the costomizations are complete, we rely on the diff schema PR check and I verified that there's no breaking change (only added field, nothing removed)

Reply:
Ultimately, these things need to be derived one way or another from the OpenAPI specification. From my read, the main classes of issues are:
1. Computed fields (not in the OpenAPI spec, but potentially could be derived from comparing get to create/update).
2. Optional fields (this should be in the OpenAPI spec).
3. Required fields (this should be in the OpenAPI spec).
4. Fields that conflict with other fields (not yet in the OpenAPI spec, but can be with `oneOf` support).
5. OpenAPI spec missing information (in the short term, we can file a ticket to the upstream team, @edwardfeng-db would be great if you can do this; in the long term, our OpenAPI generator from @hectorcast-db will address this).
6. MaxItems validation (this could also be in the OpenAPI spec).
7. Deprecated fields (this is in the OpenAPI spec but is lost in the translation to the Go SDK).
8. Enum values (these are in the OpenAPI spec, but there is no convenient way to get all enum values in the Go SDK).
9. Sensitive data (needs an extension, probably, but could be in the OpenAPI spec).
10. Suppress diff

Here's how I think we can handle this:
2: Fields are optional by default, so I'm curious whether/why we need to call `SetOptional()` for so many of them. @edwardfeng-db why is this?
3, 5: file a ticket with the Clusters team to update their OpenAPI spec so that it is complete, including missing fields and the correct annotations for required/optional.
8: We can add a method to enums to list all values; then, we could automatically compute a validator for enum fields in the resource. Let's file a ticket for this @edwardfeng-db.
1, 4, 6, 7, 9, 10: I think most of these will be most easily addressed as we try to autogenerate the provider implementation from the OpenAPI spec itself; otherwise, they require hacks to embed this metadata somehow in Go SDK struct field tags.

Reply:
Thanks. This sounds good. I'll create the follow up tickets.

For 2, there were many calls to `SetOptional()` because I was following the previous override function but I realized that a lot of these can be taken out because `omit_empty` exists on the openapi spec. I just took those out, it should look cleaner now.

Reply:
Ticket for enums: https://databricks.atlassian.net/browse/DECO-8056